File: src/contests/cg/puzzle/the-river-i-.ts
// @ts-nocheck
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

function digitSum(n: number): number {
  let s = 0;
  while (n > 0) {
    s += n % 10;
    n = Math.floor(n / 10);
  }
  return s;
}

function next(n: number): number {
  return n + digitSum(n);
}

const r1 = parseInt(readline());
const r2 = parseInt(readline());

let a = r1;
let b = r2;

while (a !== b) {
  if (a < b) {
    a = next(a);
  } else {
    b = next(b);
  }
}

console.log(a);

File: src/contests/cg/puzzle/ruby/the-river-i-.rb
# 🎮 CodinGame Puzzle - the-river-i-
# https://www.codingame.com/training/easy/the-river-i-

def digit_sum(n)
  s = 0
  while n > 0
    s += n % 10
    n /= 10
  end
  s
end

a = gets.to_i
b = gets.to_i
while a != b
  if a < b
    a += digit_sum(a)
  else
    b += digit_sum(b)
  end
end
puts a
